src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -547,8 +547,24 @@ protected override void CreateIndicesForNonTypeMembers(ITypeDefinition typeDef)
                 var deletedTypeMembers = ImmutableDictionary.CreateBuilder<IMethodDefinition, DeletedMethodDefinition>(ReferenceEqualityComparer.Instance);
                 foreach (var methodDef in deletedMethods)
                 {
+                    var overriddenMemberDef = (IMethodReference?)_changes.TryGetBaseImplementationDefinition(methodDef, out var isThisAssembly);
+
+                    uint token = 0;
+                    if (overriddenMemberDef is not null)
+                    {
+                        // Because the deleted symbols, and hence the overridden member symbol, come from a previous
+                        // generation, the normal IL emit process doesn't know how to deal with it, and will always
+                        // think its a new MemberRef row that needs to be added. We have enough information now though,
+                        // to make the correct decisions, so we just emit the raw token for the right method to call
+                        // directly.
```

Comment:
This is the explanation for why this is potentially dodgy, using the token directly.

I've put this up as a draft just to get sign off on the idea, or discuss. Obviously if this approach is okay, I'll clean up the commented out `EmitToken` call in `DeletedMethodBody` :)